src/main/scala/fm/http/LinkedHttpContentReader.scala

```diff
@@ -108,10 +108,16 @@ final class LinkedHttpContentReader(is100ContinueExpected: Boolean, head: Future
   /**
    * Write this body to a file
    */
-  def writeToFile(file: File): Future[Unit] = {
+  def writeToFile(file: File): Future[Unit] = writeToFile(file, Long.MaxValue)
+
+  def writeToFile(file: File, maxLength: Long): Future[Unit] = {
     val os: FileOutputStream = new FileOutputStream(file)
+    var bytesWritten: Long = 0
     val f: Future[Unit] = foldLeft(os){ (os, buf) =>
-      buf.readBytes(os, buf.readableBytes())
+      val readablBytes: Int = buf.readableBytes()
+      bytesWritten += readablBytes
+      buf.readBytes(os, readablBytes)
+      require(bytesWritten <= maxLength, s"Body exceeds maxLength.  Body Length (so far): $bytesWritten  Specified Max Length: $maxLength")
```

Comment:
Should the file be deleted if the `IllegalArgumentException` is thrown?

Reply:
presumably, wouldn't the file be handled by any wrapping resource or `finally`?

(e.g. in `ta-catalog` we have a `resFuture.onComplete { case _ => htmlFile.delete(); pdfFile.delete() }`)

Reply:
It looks like `file.delete()` is idempotent so it can't hurt

Reply:
Updated to delete file on failure